### DISCUSSION TITLE
Support `ds.deviceName` as an alternative to `ds.deviceId` in Data Platform connections

### DISCUSSION THREAD

**Discussion by amacneil** [foxglove/community#427] [Q&A]
Example: Customer is using BigQuery to find events of interest, and wants to generate Foxglove Studio URLs in their sql results. But they only have their `deviceName` available, not the `deviceId`.

Studio deep link URLs should support `ds.deviceName` as an alternative to `ds.deviceId` in Data Platform connections. `deviceName` is already required to be unique within an organization.

cc @defunctzombie @banisadr @jtbandes

**Comment by defunctzombie** [comment]
We can do this without any changes to apis by doing the lookup for deviceId with the name from studio. We already do a lookup to get the device with a deviceId to lookup the name, we would change this to lookup by deviceName instead so the number of queries studio makes in unchanged.
